homeassistant/components/generic/camera.py

```diff
@@ -65,9 +66,37 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a generic IP Camera."""
 
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+    _LOGGER.warning(
+        "Loading generic IP camera via configuration.yaml is deprecated, "
+        "it will be automatically imported.  Once you have confirmed correct "
+        "operation, please remove 'generic' (IP camera) section(s) from "
+        "configuration.yaml"
+    )
+    streamsource = config.get(CONF_STREAM_SOURCE)
+    config_new = {
+        CONF_NAME: config[CONF_NAME],
+        CONF_STILL_IMAGE_URL: config[CONF_STILL_IMAGE_URL].template,
+        CONF_STREAM_SOURCE: streamsource.template if streamsource is not None else None,
+        CONF_AUTHENTICATION: config.get(CONF_AUTHENTICATION),
+        CONF_USERNAME: config.get(CONF_USERNAME),
+        CONF_PASSWORD: config.get(CONF_PASSWORD),
+        CONF_LIMIT_REFETCH_TO_URL_CHANGE: config.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE),
+        CONF_CONTENT_TYPE: config.get(CONF_CONTENT_TYPE),
+        CONF_FRAMERATE: config.get(CONF_FRAMERATE),
+        CONF_VERIFY_SSL: config.get(CONF_VERIFY_SSL),
+    }
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config_new
+        )
+    )
+
 
-    async_add_entities([GenericCamera(hass, config)])
+async def async_setup_entry(hass, config, async_add_entities):
+    """Set up a generic IP Camera."""
+
+    async_add_entities([GenericCamera(hass, config.data)])
```

Comment:
`config` here is a `ConfigEntry` which has a unique id member `config.entry_id` (it generates a UUID for us since we can't provide a unique ID). See https://github.com/home-assistant/core/blob/067b5258c6bad7d020d3414449ec7aaffbd5be8a/homeassistant/config_entries.py#L141
Instead of passing in just `config.data` you can pass the whole `ConfigEntry` in to the `GenericCamera` constructor and store it and then use it to set the `unique_id` property like
```
    @property
    def unique_id(self):
        """Return unique ID."""
        return self._config_entry.entry_id

```